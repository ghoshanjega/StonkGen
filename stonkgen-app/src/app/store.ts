import { configureStore, ThunkAction, Action,getDefaultMiddleware } from '@reduxjs/toolkit';
import authenticationReducer from '../features/authentication/AuthenticationSlice';
import stocksListReducer from '../features/stocksList/StocksListSlice';
import cartReducer from '../features/cart/CartSlice';

export const store = configureStore({
  reducer: {
    authentication : authenticationReducer,
    stocksList: stocksListReducer,
    cart: cartReducer
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false
}),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
