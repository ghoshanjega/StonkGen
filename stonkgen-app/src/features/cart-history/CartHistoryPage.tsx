import { group } from "console";
import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { selectUser } from "../authentication/AuthenticationSlice";
import { ICartItemExtended } from "../cart/CartSlice";
import { DisplayStatus } from "../cart/DisplayStatus";
import { CartHistoryCards } from "./CartHistoryCards";
import { fetchCartItems, selectCartHistoryItems } from "./CartHistorySlice";

export const CartHistoryPage = () => {
  const dispatch = useDispatch();
  const cartItems = useSelector(selectCartHistoryItems);
  const user = useSelector(selectUser);
  useEffect(() => {
    if (user) {
      const sessionId = user.sessionId;
      if (sessionId) {
        dispatch(fetchCartItems({ sessionId }));
      }
    }
    return () => {};
  }, []);

  function groupBy<T, K>(list: T[], getKey: (item: T) => K) {
    const map = new Map<K, T[]>();
    list.forEach((item) => {
      const key = getKey(item);
      const collection = map.get(key);
      if (!collection) {
        map.set(key, [item]);
      } else {
        collection.push(item);
      }
    });
    return Array.from(map.values());
  }

  return (
    <div className="container my-5">
     <CartHistoryCards cartItems={cartItems} />
    </div>
  );
};
