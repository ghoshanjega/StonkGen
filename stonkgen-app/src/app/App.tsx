import React, { useContext } from "react";
import { useSelector } from "react-redux";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { Footer } from "../components/Footer";
import { Header } from "../components/Header";
import { AdminPage } from "../features/admin/AdminPage";

import { AuthenticationPage } from "../features/authentication/AuthenticationPage";
import { selectUser } from "../features/authentication/AuthenticationSlice";
import { CartHistoryPage } from "../features/cart-history/CartHistoryPage";
import { CartPage } from "../features/cart/CartPage";
import { StocksListPage } from "../features/stocksList/StocksListPage";

import "./App.css";

function App() {
  const user = useSelector(selectUser);
  if (!user) {
    return (
      <div>
        <AuthenticationPage />
      </div>
    );
  }
  return (
    <div className="wrapper">
      <BrowserRouter>
        <Header />
        <Switch>
          <Route path="/orders">
            <CartPage />
          </Route>
          <Route path="/order-history">
            <CartHistoryPage />
          </Route>
          <Route path="/admin">
            <AdminPage />
          </Route>
          <Route path="/login">
            <AuthenticationPage />
          </Route>
          <Route path="/">
            <StocksListPage />
          </Route>
        </Switch>
      </BrowserRouter>
      <Footer />
    </div>
  );
}

export default App;
