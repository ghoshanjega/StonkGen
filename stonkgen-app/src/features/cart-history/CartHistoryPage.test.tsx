import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { store } from "../../app/store";

import { CartHistoryPage } from './CartHistoryPage'

describe("Cart History Page Component", () => {
    let component: renderer.ReactTestRenderer;
    beforeEach(() => {
      store.dispatch = jest.fn();
  
      component = renderer.create(
        <Provider store={store}>
          <CartHistoryPage />
        </Provider>
      );
    });

    it("should render the snapshot", () => {
        expect(component).toMatchSnapshot();
      });

    
})