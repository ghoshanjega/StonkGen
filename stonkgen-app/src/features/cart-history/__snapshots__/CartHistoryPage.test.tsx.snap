// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart History Page Component should render the snapshot 1`] = `
<div
  className="container my-5"
>
  <div
    className="jumbotron jumbotron-fluid"
  >
    <div
      className="container"
    >
      <p
        className="lead text-center"
        data-test="nothing"
      >
        Nothing to show yet
      </p>
    </div>
  </div>
</div>
`;

exports[`Header Component should render the header snapshot 1`] = `
<div
  className="container my-5"
>
  <div
    className="jumbotron jumbotron-fluid"
  >
    <div
      className="container"
    >
      <p
        className="lead text-center"
        data-test="nothing"
      >
        Nothing to show yet
      </p>
    </div>
  </div>
</div>
`;
